core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -161,6 +165,83 @@ static ClientRequestContextBuilder builder(RpcRequest request, URI uri) {
         return new ClientRequestContextBuilder(request, uri);
     }
 
+    /**
+     * Returns the {@link ServiceRequestContext} whose {@link Service} invokes the {@link Client}
+     * {@link Request} which created this {@link ClientRequestContext}, or {@code null} if this client request
+     * was not made in the context of a server request.
+     */
+    @Nullable
+    ServiceRequestContext rootContext();
```

Comment:
Sholuld we call it `root()` or not?

Reply:
Well, either is fine I guess. You think `root()` fits better?

Reply:
@anuraaga Do you have an idea on this? 😄

Reply:
`root()` seems slightly better to me too.

Reply:
Let me use `root()`, then. Thanks!